go/observability/observability.go

```diff
@@ -0,0 +1,29 @@
+package observability
+
+import (
+	"log"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"go.opencensus.io/plugin/ocgrpc"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+func ConfigureStackdriver(gcpProject string) {
+	if err := view.Register(ocgrpc.DefaultClientViews...); err != nil {
+		log.Fatalf("Failed to register ocgrpc client views: %v", err)
+	}
+
+	sd, err := stackdriver.NewExporter(stackdriver.Options{
+		ProjectID: gcpProject,
+	})
+	if err != nil {
+		log.Fatalf("Failed to create Stackdriver exporter: %v", err)
+	}
+	defer sd.Flush()
```

Comment:
What does `Flush` do? And if it does what I think it does, I don't we need to defer it.
(Deferred functions run when the current function returns, I'm not sure what we want to flush here)

And does it happen periodically afterward? Or we will need a goroutine to call it.

Reply:
`Flush` sends any buffered traces/metrics to stackdriver. It is intended to be used when your program is exiting to ensure data is not lost. You're right, I misapplied the OpenCensus example (and misunderstand what `defer` means).